packages/devtools_app/lib/src/banner_messages.dart

```diff
@@ -254,6 +254,50 @@ Relaunch your application with the '--profile' argument, or ''',
   }
 }
 
+class ProviderUnknownErrorBanner {
+  const ProviderUnknownErrorBanner({@required this.screenId});
+
+  final String screenId;
+
+  BannerMessage build(BuildContext context) {
+    return _BannerError(
+      key: Key('ProviderUnknownErrorBanner - $screenId'),
+      screenId: screenId,
+      textSpans: const [
+        TextSpan(
+          text: '''
+The devtool failed to connect with package:provider.
+
+This could be caused by an outdated version of package:provider. Make sure that you are using a version >=5.0.0.''',
```

Comment:
Instead of guessing, can you tweak package:provider so you can tell exactly which version it is from code?
For example, you could check for the presence of an API not present before 5.0.0 using EvalOnLibrary or store a string in the library that lists the version.